pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+        class_for_inst = self.classes[namespace][new_inst.classname]
+
+        # If property not in instance, add it and use default value from
+        # class
+        for cprop_name in class_for_inst.properties:
+            if cprop_name not in new_inst:
+                default_value = class_for_inst.properties[cprop_name]
+                new_inst[cprop_name] = default_value
+
+        # Exception if property in instance but not class
+        for iprop in new_inst:
+            if iprop not in class_for_inst.properties:
+                raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                               'Property %s in instance but not in class %s'
+                               % (iprop, class_for_inst))
+
+        # if namespace in instance.path use it.
+        path = self.create_instance_path(class_for_inst, new_inst, namespace)
+        new_inst.path = path
+        try:
+            for inst in self.instances[namespace]:
+                if inst.path == new_inst.path:
+                    raise CIMError(CIM_ERR_ALREADY_EXISTS,
+                                   'Instance %s exists in namespace %s' %
+                                   (new_inst.path, namespace))
+            self.instances[namespace].append(new_inst)
+        except KeyError:
+            self.instances[namespace] = [new_inst]
+        # Create instance returns model path, path relative to namespace
+        # TODO per spec. path returned if any keys are dynamically allocated.
+        return self._make_tuple([copy.deepcopy(path)])
+
+    def _fake_modifyinstance(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`~pywbem.WBEMConnection.CreateInstance`
+
+        Create a CIM instance in the local repository of this class.
+
+        Exceptions:
+            CIMError CIM_ERR_ALREADY_EXISTS, CIM_ERR_INVALID_CLASS
+        """
+        modified_instance = params['ModifiedInstance']
+        # class_name = modified_instance.classname
+        # TODO class must exist
+
+        inst_name = modified_instance.path
+
+        # classname in instance and path must match
+
+        orig_inst = self._get_instance(inst_name, namespace, **params)
+
+        # ignore IncludeQualifiers as it has been deprecated and its use
+        # by servers is not determinate.
+
+        property_list = params['PropertyList']
+
+        if property_list is not None:
+            self._filter_properties(modified_instance, property_list)
+            # TODO No to following
+            modified_instance.update(modified_instance.path)
+
+        # self._set_instance(cim_class=cimClass
+        #                  instance=modifiedInstance,
+        #                  previous_instance=previousInstance,
+        #                  )
+
+        # find instance in list
+        for inst in self.instances[namespace]:
+            if inst.path == orig_inst.path:
+                inst = modified_instance
+                return
```

Comment:
Something is missing here: Where is the instance in the repo being midified?

Reply:
TODO:

Reply:
Done.  However, testing broken on modifyinstance so I have TODO on the testing